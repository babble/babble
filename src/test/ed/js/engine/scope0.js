
/**
*    Copyright (C) 2008 10gen Inc.
*  
*    This program is free software: you can redistribute it and/or  modify
*    it under the terms of the GNU Affero General Public License, version 3,
*    as published by the Free Software Foundation.
*  
*    This program is distributed in the hope that it will be useful,
*    but WITHOUT ANY WARRANTY; without even the implied warranty of
*    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*    GNU Affero General Public License for more details.
*  
*    You should have received a copy of the GNU Affero General Public License
*    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

var a = 1;
var b = 2;

function foo(){
    return a + b;
}

print( "3=" + foo() );
a = 2;
b = 3;
print( "5=" + foo() );

var v = 13;
function bar(){
    v = v + 2;
    return v;
}
print( bar() );
print( bar() );
print( v );

function asdads( a ){
    a = 5;
    var b = a + 2;
    return a + b;
}

    
