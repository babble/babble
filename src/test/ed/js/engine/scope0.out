3=3.0
5=5.0
